### ISSUE TITLE
Running bash within a fish shell breaks command history

### ISSUE THREAD

**Issue by btglr** [fish-shell/fish-shell#8349]
When running bash within a fish shell and executing one or more commands afterwards, the commands are appended to the fish_history file. Similarly to #6032, the `history` command then isn't able to display most of the history. Weirdly, some of the entries before the malformed line do show in the `history` command.

# Recreation steps

- Run `bash`
- Execute any command in the bash shell
- Ctrl-D

# Console output

```bash
╭─pc@pc in ~
╰─λ bash

╭─pc@pc in ~ took 6ms
╰─λ echo $SHELL
/bin/bash

╭─pc@pc in ~ took 3ms
╰─λ
exit

╭─pc@pc in ~ took 37s
╰─λ tail ~/.local/share/fish/fish_history
- cmd: bash
when: 1634199415
echo $SHELL
- cmd: tail ~/.local/share/fish/fish_history
when: 1634199457
paths:
- ~/.local/share/fish/fish_history

╭─pc@pc in ~ took 1ms
╰─λ history
2021-10-14 10:17:37 tail ~/.local/share/fish/fish_history
2021-10-14 10:16:55 bash
2021-10-14 10:12:53 echo $TERM
2021-10-14 10:12:51 uname -a
2021-10-14 10:12:34 echo $version
2021-10-14 10:12:05 fish --version
```

# System information

```bash
╭─pc@pc in ~
╰─λ fish --version
fish, version 3.3.1-393-g86a442fb6

╭─pc@pc in ~
╰─λ uname -a
Linux pc-xps139370 5.14.9-zen2-1-zen #1 ZEN SMP PREEMPT Fri, 01 Oct 2021 19:03:23 +0000 x86_64 GNU/Linux

╭─pc@pc in ~ took 1ms
╰─λ echo $TERM
xterm-256color
```

**Comment by faho** [comment]
I can't reproduce this, and it seems like it would be caused by your configuration.

Please check your config.fish and any functions and third-party "plugins" you might have, like the issue template asks.

**Comment by btglr** [comment]
> I can't reproduce this, and it seems like it would be caused by your configuration.
> 
> Please check your config.fish and any functions and third-party "plugins" you might have, like the issue template asks.

When using `sh -c 'env HOME=$(mktemp -d) fish'` and doing the steps I mentioned, the commands executed in the bash subshell appear in the ~/.local/share/fish/fish_history file anyway, and not in the temporary one.
Trying to determine if it comes from my user configuration, I started bash without any profile using `bash --noprofile --norc`. The problem still occurred and that's when I also noticed that using `history` shows the fish_history file as plain text:

```
491  - cmd: history
492    when: 1634202776
493  - cmd: lvim ~/.local/share/fish/fish_history
494    when: 1634202798
495    paths:
496      - ~/.local/share/fish/fish_history
497  - cmd: cat /etc/profile
498    when: 1634202890
499    paths:
500      - /etc/profile
501  history
```
The last line was executed in bash and gets added to the ~/.local/share/fish/fish_history file even with the temporary $HOME.

I then ran bash within the temporary fish shell using an empty environment: `env -i bash --norc --noprofile`. This time, the history command worked as expected and only showed bash history, and nothing was appended to the ~/.local/share/fish/fish_history file:

```
97  echo $SHELL
98  history
99  history
```

It seems it indeed comes from my system's bash configuration and not from fish. I don't see anything wrong with /etc/profile and /etc/bash.bashrc so I'm not sure where the problem lies.
